### ISSUE TITLE
Url with '@' symbol needs to be parsed as valid

### ISSUE THREAD

**Issue by michellai** [nozbzh/urlshortener#1]
Currently an '@' symbol (as in maps links, or twitter links) will be recognized as invalid.

Expected:
Shortened Url is returned

**Comment by nozbzh** [comment]
It seems like big URLs can't be processed by the regexp somehow.

A maps link or a google search link will make the regexp try to match indefinitely without failing or passing the test.

URL below is a good test case to work on:

https://www.google.com/search?sourceid=chrome-psyapi2&ion=1&espv=2&es_th=1&ie=UTF-8&q=google%20maps%20link%20regexp&oq=google%20maps%20link%20regexp&aqs=chrome..69i57j69i64l3.9387j0j7

**Comment by nozbzh** [comment]
Maybe the solution is to simply be very liberal when validating the format...

goo.gl and ow.ly give me a link for "fdss" or any random string that is not a URL at all.

They both reject "veas45$%&*&$%^" as a valid URL though, so there is some sort of check.

**Comment by nozbzh** [comment]
Turns out there's a built in regexp for that!

I'll paste it just for you @michellai :)

```
/
    ([a-zA-Z][\-+.a-zA-Z\d]*):                           (?# 1: scheme)
    (?:
       ((?:[\-_.!~*'()a-zA-Z\d;?:@&=+$,]|%[a-fA-F\d]{2})(?:[\-_.!~*'()a-zA-Z\d;\/?:@&=+$,\[\]]|%[a-fA-F\d]{2})*)                    (?# 2: opaque)
    |
       (?:(?:
         \/\/(?:
             (?:(?:((?:[\-_.!~*'()a-zA-Z\d;:&=+$,]|%[a-fA-F\d]{2})*)@)?        (?# 3: userinfo)
               (?:((?:(?:[a-zA-Z0-9\-.]|%\h\h)+|\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}|\[(?:(?:[a-fA-F\d]{1,4}:)*(?:[a-fA-F\d]{1,4}|\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})|(?:(?:[a-fA-F\d]{1,4}:)*[a-fA-F\d]{1,4})?::(?:(?:[a-fA-F\d]{1,4}:)*(?:[a-fA-F\d]{1,4}|\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}))?)\]))(?::(\d*))?))? (?# 4: host, 5: port)
           |
             ((?:[\-_.!~*'()a-zA-Z\d$,;:@&=+]|%[a-fA-F\d]{2})+)                 (?# 6: registry)
           )
         |
         (?!\/\/))                           (?# XXX: '\/\/' is the mark for hostport)
         (\/(?:[\-_.!~*'()a-zA-Z\d:@&=+$,]|%[a-fA-F\d]{2})*(?:;(?:[\-_.!~*'()a-zA-Z\d:@&=+$,]|%[a-fA-F\d]{2})*)*(?:\/(?:[\-_.!~*'()a-zA-Z\d:@&=+$,]|%[a-fA-F\d]{2})*(?:;(?:[\-_.!~*'()a-zA-Z\d:@&=+$,]|%[a-fA-F\d]{2})*)*)*)?                    (?# 7: path)
       )(?:\?((?:[\-_.!~*'()a-zA-Z\d;\/?:@&=+$,\[\]]|%[a-fA-F\d]{2})*))?                 (?# 8: query)
    )
    (?:\#((?:[\-_.!~*'()a-zA-Z\d;\/?:@&=+$,\[\]]|%[a-fA-F\d]{2})*))?                  (?# 9: fragment)
  /x
```

**Comment by michellai** [comment]
Mon Hero

**Comment by nozbzh** [comment]
Closed issue because it has been fixed in commit 1c85cd99fa8f6b621bedc12ccd366f9b762be542
